spyder_env_manager/spyder/widgets/packages_table.py

```diff
@@ -126,59 +112,110 @@ def row(self, row_num):
-        self.source_model = EnvironmentPackagesModel(self, text_color=text_color)
+    sig_action_context_menu = Signal(str, dict)
+
+    def __init__(self, parent):
+        super().__init__(parent)
+        # Setup table model
+        self.source_model = EnvironmentPackagesModel(self)
         self.setModel(self.source_model)
+
+        # Setup table
         self.setSelectionBehavior(QAbstractItemView.SelectRows)
         self.setSelectionMode(QAbstractItemView.SingleSelection)
-        self.setSortingEnabled(True)
         self.setEditTriggers(QAbstractItemView.AllEditTriggers)
         self.verticalHeader().hide()
+        self.horizontalHeader().setStretchLastSection(True)
         self.load_packages(False)
 
+    def selection(self, action, package_info):
+        """Update selected row."""
+        self.sig_action_context_menu.emit(action, package_info)
+
+    def get_package_info(self, index):
+        return self.source_model.packages[index]
+
+    def load_packages(self, only_requested=False, packages=None):
+        #     packages = [
+        #         {
+        #             "name": "package name",
+        #             "description": "package description",
+        #             "version": "0.0.1",
+        #             "requested": False,
+        #         },
+        #     ]
+        if packages:
+            self.source_model.all_packages = packages
+        if not packages and self.source_model.all_packages:
+            packages = self.source_model.all_packages
+        if packages:
+            if only_requested:
+                packages = list(filter(lambda package: package["requested"], packages))
+            for idx, package in enumerate(packages):
+                package["index"] = idx
+            packages_map = {package["name"]: package for package in packages}
+
+            self.source_model.packages = packages
+            self.source_model.packages_map = packages_map
+            self.source_model.reset()
+            self.resizeColumnToContents(NAME)
+
+    def next_row(self):
+        """Move to next row from currently selected row."""
+        row = self.currentIndex().row()
+        rows = self.source_model.rowCount()
+        if row + 1 == rows:
+            row = -1
+        self.selectRow(row + 1)
+
+    def previous_row(self):
+        """Move to previous row from currently selected row."""
+        row = self.currentIndex().row()
+        rows = self.source_model.rowCount()
+        if row == 0:
+            row = rows
+        self.selectRow(row - 1)
+
     def contextMenuEvent(self, event):
         """Setup context menu"""
         row = self.rowAt(event.pos().y())
         packages = self.source_model.packages
-        if not packages[row]["dependence"]:
-            self.update_action = create_action(
-                self, _("Update package(s)"), triggered=self.selection
+        if packages and packages[row]["requested"]:
+            update_action = create_action(
```

Comment:
Perhaps it'd a good idea if this widget inherits from `SpyderWidgetMixin`, so it can `self.create_action` instead of `create_action` directly, which is a low level function.